lib/datadog/core/configuration/option.rb

```diff
@@ -191,11 +189,33 @@ def coerce_env_variable(value)
           when :string, NilClass
             value
           else
-            raise ArgumentError,
+            raise InvalidDefinitionError,
               "The option #{@definition.name} is using an unsupported type option for env coercion `#{@definition.type}`"
           end
         end
 
+        def coerce_env_var_int(value)
+          # Strict integer parsing
+          Integer(value, 10)
+        rescue ArgumentError => e
+          # Try also parsing as a whole floating point numbers
+          begin
+            f = Float(value)
+
+            # Check for whole numbers
+            raise ArgumentError, "#{value} is not a whole number" unless f.truncate == f
+
+            f
```

Comment:
Minor: Output an `Integer` in the end?

```suggestion
            f.to_i
```